iwyu_preprocessor.cc

```diff
@@ -1087,4 +1126,21 @@ bool IwyuPreprocessorInfo::ForwardDeclareIsInhibited(
       ContainsKey(*inhibited_forward_declares, normalized_symbol_name);
 }
 
+bool IwyuPreprocessorInfo::ForwardDeclareInKeepRange(
+    const clang::SourceLocation& loc) const {
+  // Check if the entry exists, shortcircuit if not
+  const clang::FileEntry* ref = GetFileEntry(loc);
+  if (keep_location_ranges_.find(ref) == keep_location_ranges_.end()) {
+    return false;
+  }
+
+  const clang::SourceRange mRange{loc};
```

Comment:
We don't use camelCase in IWYU, but rather hacker_case. Should this maybe be called `decl_range`?